conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/ApiDataHelpers.scala

```diff
@@ -0,0 +1,121 @@
+import tech.cryptonomic.conseil.common.util.Conversion.Syntax._
+
+/** Trait with methods for converting from data types to Json */
+trait ApiDataHelpers extends AkkaEndpoints with AkkaJsonSchemaEntities with ApiDataJsonSchemas {
+
+  /** Function validating request for the query endpoint */
+  def defaultValidated[A](
+      response: A => Route,
+      invalidDocs: Documentation
+  ): Either[List[QueryValidationError], A] => Route = {
+    case Left(errors) =>
+      complete(StatusCodes.BadRequest -> s"Errors: \n${errors.mkString("\n")}")
+    case Right(QueryResponseWithOutput(queryResponse, OutputType.csv)) =>
+      complete(HttpEntity.Strict(ContentTypes.`text/csv(UTF-8)`, ByteString(queryResponse.convertTo[String])))
+    case Right(QueryResponseWithOutput(queryResponse, OutputType.sql)) =>
+      complete(HttpEntity.Strict(ContentTypes.`text/plain(UTF-8)`, ByteString(queryResponse.head("sql").get.toString)))
+    case Right(success) =>
+      response(success)
+  }
+
+  private lazy val anyEncoder: Encoder[Any] =
+    Encoder.instance(x => defaultAnyEncoder orElse customAnyEncoder applyOrElse (x, fallbackAnyEncoderValue))
+
+  /** Represents the function, that is going to encode the most common types */
+  private def defaultAnyEncoder: PartialFunction[Any, Json] = {
+    case x: java.lang.String => Json.fromString(x)
+    case x: java.lang.Integer => Json.fromInt(x)
+    case x: java.lang.Long => Json.fromLong(x)
+    case x: java.sql.Timestamp => Json.fromLong(x.getTime)
+    case x: java.lang.Boolean => Json.fromBoolean(x)
+    case x: scala.collection.immutable.Vector[Any] =>
+      x.map(_.asJson(anyEncoder)).asJson //Due to type erasure, a recursive call is made here.
+    case x: java.math.BigDecimal => Json.fromBigDecimal(x)
+  }
+
+  /** Represents the function, that is going to encode the blockchain specific data types */
+  protected def customAnyEncoder: PartialFunction[Any, Json]
+
+  /** Represents the function, that is going to return the fallback value (when any other encoder will fail) */
+  private def fallbackAnyEncoderValue: Any => Json = x => Json.fromString(x.toString)
+
+  /** JSON schema implementation for Any */
+  implicit def anySchema: JsonSchema[Any] = new JsonSchema[Any] {
+    override def encoder: Encoder[Any] = anyEncoder
+
+    override def decoder: Decoder[Any] =
+      (c: HCursor) => {
+        // without this check strings are deserialized in double quotes for example "String" instead of String
+        if (c.value.isString) {
+          Right(c.value.asString.get)
+        } else {
+          Right(c.value)
+        }
+      }
+  }
+
+  /** Query response JSON schema implementation */
+  implicit def queryResponseSchema: JsonSchema[QueryResponse] =
+    new JsonSchema[QueryResponse] {
+      override def encoder: Encoder[QueryResponse] =
+        (a: QueryResponse) =>
+          Json.obj(a.map(field => (field._1, field._2.map(_.asJson(anyEncoder)).getOrElse(Json.Null))).toList: _*)
```

Comment:
For the sake of reading better, what would you think of changing this to a pattern matching input like

```scala
a.map { case (fieldName, fieldValue) => 
  fieldName -> fieldValue.map(...).getOrElse(...)
}.toList
```

Reply:
Make sense